Artur_Aghajanyan/21_05_2021/tic_tac_toe.sh

```diff
@@ -0,0 +1,212 @@
+  read withComputer
+  echo -e "\e[33mChoose figure${END}"
+  echo -e "\e[4mCommands${END}: ${BLUE}(X)X${END}   ${RED}(O)O${END}"
+  res=true
+  while $res; do
+    read figure
+    if [[ $figure == "X" ]] || [[ $figure == "x" ]]
+    then
+      play=$player_1
+      res=flase
+    elif [[ $figure == "O" ]] || [[ $figure == "o" ]]
+    then
+      play=$player_2
+      res=flase
+    else
+      echo -e "${RED}Incorrect Figure!${END}"
+    fi
+  done
+}
+
+
+printBoard () {
+  clear
+  echo -e $WHITE_B
+  echo -e " ${moves[0]} | ${moves[1]} | ${moves[2]} "
+  echo "-----------"
+  echo -e " ${moves[3]} | ${moves[4]} | ${moves[5]} "
+  echo "-----------"
+  echo -e " ${moves[6]} | ${moves[7]} | ${moves[8]} "
+  echo -e $END
+  echo "============="
+}
+
+playerPick(){
+  if [[ $(($turn % 2)) == 0 ]]
+  then
+    echo -en "${YELLOW}PLAYER 2 PICK A SQUARE: ${END}"
+  else
+    echo -en "${YELLOW}PLAYER 1 PICK A SQUARE: ${END}"
+  fi
+  if [ $withComputer == "c" ] || [ $withComputer == "C" ] || [ $withComputer == "copmuter" ] || [ $withComputer == "Computer" ]
+  then
+    if [[ $(($turn % 2)) == 0 ]]
+    then
+      square=$((1 + $RANDOM % 9))
+      while [[ ! $square =~ ^-?[0-9]+$ ]] && [[ ! $space =~ ^[0-9]+$  ]]
+      do
+        space=${moves[($square -1)]} 
+        echo $square
+        sq=$(expr $square - 1)
+        echo $(expr $sq)
+        echo ${move[($square -1)]}
+        sleep 1
+        square=$((1 + $RANDOM % 9))
+      done
+    else
+      read square
+    fi
+  else
+    read square
+  fi
+
+  space=${moves[($square -1)]} 
+  if [[ ! $square =~ ^-?[0-9]+$ ]] || [[ ! $space =~ ^[0-9]+$  ]]
+  then
+    echo "Not a valid square."
+    playerPick
+  else
+    
+    moves[($square -1)]=$play
+    if [ $play == $player_1 ]
+    then 
+      play=$player_2
+    else
+      play=$player_1
+    fi
+    ((turn=turn+1))
+  fi
+  space=${moves[($square-1)]} 
+}
+
+checkMatch() {
+  if  [ ${moves[$1]} == ${moves[$2]} ] && [ ${moves[$2]} == ${moves[$3]} ]; then
+    game_on=false
+  fi
+  
+  if [ $game_on == false ]; then
+    if [ ${moves[$1]} == $player_1 ];then
+      echo -e "${BLUE}The winners are the Xs!${END}"
+      sleep 3
+      clear
+      echo -e "\e[33mDo you want to play again?${END}"
+      echo -e "\e[4mCommands${END}: ${GREEN}(Y)Yes${END}   ${RED}(N or any other key)End${END}"
+      read command
+      if [ $command == "yes" ] || [ $command == "y" ] || [ $command == "Y" ] || [ $command == "Yes" ]
+      then 
+        playAgain
+      else
+        exit 
+      fi
+    else
+      echo -e "${RED}The winners are the Os!${END}"
+      sleep 3
+      clear
+      echo -e "\e[33mDo you want to play again?${END}"
+      echo -e "\e[4mCommands${END}: ${GREEN}(Y)Yes${END}   ${RED}(N or any other key)End${END}"
+      read command
+      if [ $command == "yes" ] || [ $command == "y" ] || [ $command == "Y" ] || [ $command == "Yes" ]
+      then 
```

Comment:
![image](https://user-images.githubusercontent.com/80317835/119309747-c19b9700-bc7f-11eb-9f56-b3682da88a01.png)
getting the error above